### PR TITLE
Fix the document of UseCaseContext.constructor

### PR DESCRIPTION
**Pull request by saneyuki**
Fix https://github.com/almin/almin/issues/85

### CODE REVIEW

File: src/UseCaseContext.ts [hunk 1]
```diff
@@ -16,18 +16,16 @@ export default class UseCaseContext {
     dispatcher: UseCase;
 
     /**
-     * @param   [dispatcher]    dispatcher is Dispatcher or parent UseCase.
-     * @public
+     * @param   dispatcher
+     *  The parent UseCase.
      */
     constructor(dispatcher: UseCase) {
```
Comment (saneyuki) [inline src/UseCaseContext.ts]: @azu I forget to ask you about the following question to you.

1. In the document, this `dispatcher` arguments is explained as "parent use case", and this value is passed as `UseCaseExecutorArgs.parent` in `UseCaseExecutorArgs.useCase()`.
2. in jsdoc, this `dispatcher` is marked as optional arguments.

By these things, I worry whether we should make it an optional argument by adding `null` default value like:

```typescript
constructor(dispatcher: UseCase = null);
```

How about do you think?

Comment (azu) [inline src/UseCaseContext.ts]: > in jsdoc, this dispatcher is marked as optional arguments.

This is wrong. `dispatcher` of the UseCaseContext's constructor is require.
([This commit](in/pull/91/commits/860dea85fe7bb9d9f36bdb679c8c5caa90a4a4a5)  is no problem)

Insteadof this, We can add `null` default value to `UseCaseExecutor` constructor.

```ts
    constructor({
        useCase,
        parent = null, // <= 
        dispatcher
    }: UseCaseExecutorArgs)
```

summarized

- `new UseCaseExecutor({ useCase, dispatcher, parent = null })` from `Context`.
- `new UseCaseExecutor({ useCase, dispatcher, parent = dispatcher})` from `UseCaseContext`.

Comment (saneyuki) [inline src/UseCaseContext.ts]: > Insteadof this, We can add null default value to UseCaseExecutor constructor.

`UseCaseExecutor.parent` has already been `UseCase | null`. And there is no error at the HEAD of this branch. Really do we need change to what you suggested?

Comment (azu) [inline src/UseCaseContext.ts]: > Really do we need change to what you suggested?

Ok. it is nonessary.

**Comment by azu** [comment]
lgtm

**Comment by saneyuki** [comment]
@azu could you merge this? I don't have a merge permission.

**Comment by azu** [comment]
fmm, I've merged this.

Additionaly, I've changed your role.
saneyuki s access for almin 2017-02-22 16-26-56

**Pull request merged**